src/main/java/org/zalando/nakadi/service/subscription/zk/AbstractZkSubscriptionClient.java

```diff
@@ -0,0 +1,480 @@
+package org.zalando.nakadi.service.subscription.zk;
+
+import com.google.common.base.Charsets;
+import static com.google.common.base.Charsets.UTF_8;
+import com.google.common.collect.Lists;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.recipes.cache.NodeCache;
+import org.apache.curator.framework.recipes.locks.InterProcessSemaphoreMutex;
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.KeeperException;
+import org.apache.zookeeper.data.Stat;
+import org.echocat.jomon.runtime.concurrent.RetryForSpecifiedCountStrategy;
+import static org.echocat.jomon.runtime.concurrent.Retryer.executeWithRetry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.zalando.nakadi.domain.EventTypePartition;
+import org.zalando.nakadi.exceptions.NakadiRuntimeException;
+import org.zalando.nakadi.exceptions.ServiceUnavailableException;
+import org.zalando.nakadi.exceptions.UnableProcessException;
+import org.zalando.nakadi.exceptions.runtime.OperationInterruptedException;
+import org.zalando.nakadi.exceptions.runtime.OperationTimeoutException;
+import org.zalando.nakadi.exceptions.runtime.RequestInProgressException;
+import org.zalando.nakadi.exceptions.runtime.ZookeeperException;
+import org.zalando.nakadi.service.subscription.model.Session;
+import org.zalando.nakadi.view.SubscriptionCursorWithoutToken;
+
+public abstract class AbstractZkSubscriptionClient implements ZkSubscriptionClient {
+    private static final String STATE_INITIALIZED = "INITIALIZED";
+    private static final int COMMIT_CONFLICT_RETRY_TIMES = 5;
+    static final String NODE_TOPOLOGY = "/topology";
+
+    private final String subscriptionId;
```

Comment:
Actually, methods of this class is based on the subscription id. Having it as ctr arg makes the code cleaner, since you won't pass the same arg `subscription.getId()` for every call of this class. There is a subscription client factory class which is spring bean, so you can inject it and create ZkSubscriptionClient. 
In general, I agree that we should use feature of the framework we use, but in this case it handy to have it implemented this way. So, probably, for now we can leave as it is and refactor it once we get rid of OldZK.. Sorry, @v-stepanov :)